hermes-management/src/main/java/pl/allegro/tech/hermes/management/infrastructure/kafka/service/KafkaRawMessageReader.java

```diff
@@ -1,64 +1,66 @@
 package pl.allegro.tech.hermes.management.infrastructure.kafka.service;
 
-import kafka.api.FetchRequestBuilder;
-import kafka.javaapi.FetchResponse;
-import kafka.javaapi.consumer.SimpleConsumer;
-import kafka.message.MessageAndOffset;
+import java.time.Duration;
+import java.util.Collections;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.common.TopicPartition;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import pl.allegro.tech.hermes.common.kafka.KafkaTopic;
-import pl.allegro.tech.hermes.common.kafka.SimpleConsumerPool;
+import pl.allegro.tech.hermes.common.kafka.KafkaConsumerPool;
 import pl.allegro.tech.hermes.management.domain.topic.SingleMessageReaderException;
 
-import java.nio.ByteBuffer;
-
 import static java.lang.String.format;
 
 public class KafkaRawMessageReader {
 
     private static final Logger logger = LoggerFactory.getLogger(KafkaRawMessageReader.class);
 
-    private final SimpleConsumerPool simpleConsumerPool;
+    private final KafkaConsumerPool consumerPool;
+    private final int pollTimeoutMillis;
 
-    public KafkaRawMessageReader(SimpleConsumerPool simpleConsumerPool) {
-        this.simpleConsumerPool = simpleConsumerPool;
+    public KafkaRawMessageReader(KafkaConsumerPool consumerPool,
+                                 int pollTimeoutMillis) {
+        this.consumerPool = consumerPool;
+        this.pollTimeoutMillis = pollTimeoutMillis;
     }
 
-    public byte[] readMessage(KafkaTopic topic, int partition, long offset) {
-        FetchResponse fetchResponse = fetch(topic, partition, offset);
-        for (MessageAndOffset messageAndOffset : fetchResponse.messageSet(topic.name().asString(), partition)) {
-            if (messageAndOffset.offset() == offset) {
-                return readPayloadAsBytes(messageAndOffset);
+    byte[] readMessage(KafkaTopic topic, int partition, long offset) {
+        KafkaConsumer<byte[], byte[]> kafkaConsumer = consumerPool.get(topic, partition);
+        TopicPartition topicPartition = new TopicPartition(topic.name().asString(), partition);
+
+        try {
+            kafkaConsumer.assign(Collections.singleton(topicPartition));
+            kafkaConsumer.seek(topicPartition, offset);
+            ConsumerRecords<byte[], byte[]> records = kafkaConsumer.poll(Duration.ofMillis(pollTimeoutMillis));
+            for (ConsumerRecord<byte[], byte[]> record : records.records(topicPartition)) {
+                if (record.offset() == offset) {
+                    return record.value();
+                }
+                logger.info("Found an old offset: {} Expecting: {}", record.offset(), offset);
```

Comment:
Do we need to log it as an info, maybe debug?

Reply:
It almost never happens and if it happens it is good to know immediately